core/controllers/topic_editor.py

```diff
@@ -84,6 +88,48 @@ def post(self, topic_id):
         })
 
 
+class TopicEditorQuestionHandler(base.BaseHandler):
+    """Manages the creation of a question and receiving of all question
+    summaries for display in topic editor page.
+    """
+
+    @acl_decorators.can_edit_topic
+    def get(self, topic_id):
+        """Handles GET requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+
+        topic = topic_services.get_topic_by_id(topic_id)
+        skill_ids = topic.get_all_skill_ids()
+
+        question_summaries = (
+            question_services.get_question_summaries_linked_to_skills(skill_ids)
```

Comment:
Is there any paging for this? I think it's going to grow too large too quickly so I think paging is necessary.

Happy for that to be added in separate PR (to unblock @nithusha21) but it should happen soon. If you decide to go this route then please file an issue.

(In general, think about scale, so we don't have to rewrite all this a second time.)

Reply:
Even if paging is there, shouldn't it be added in the frontend? or should this be returned in batches each time a new page is selected (in the UI) from the backend itself?

Reply:
I would say batches (and look at e.g. things like https://cloud.google.com/appengine/docs/standard/python/ndb/queries#cursors). You don't want to pull 10000 entities in the backend and then only use 5 of them each time.

Reply:
Will add in this PR itself.